### PR TITLE
openiothub-server: rename from server-go

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by dawidd6** [comment]
IMO this formula is named very generically. Could we rename it to something like `openiothub-server`? Thoughts?

**Comment by chenrui333** [comment]
make sense.

**Comment by bayandin** [comment]
Can we build it with the latest `go`?

```
  # Bump to 1.18 on the next release, if possible.
  depends_on "go@1.17" => :build
```

File: Formula/openiothub-server.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      -s -w
+      -X main.version=#{version}
+      -X main.commit=#{Utils.git_head}
+      -X main.date=#{time.iso8601}
+      -X main.builtBy=homebrew
+    ]
+
+    (etc/"server-go").mkpath
+    system "go", "build", "-mod=vendor", *std_go_args(ldflags: ldflags)
+    etc.install "server-go.yaml" => "server-go/server-go.yaml"
```
Comment (SMillerDev) [inline Formula/openiothub-server.rb:23]:
```suggestion
    system "go", "build", "-mod=vendor", *std_go_args(ldflags: ldflags)
    bin.install_symlink bin/"openiothub-server" => "server-go"
    etc.install "server-go.yaml" => "server-go/server-go.yaml"
```

File: Formula/openiothub-server.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  service do
+    run [opt_bin/"server-go", "-c", etc/"server-go/server-go.yaml"]
+    keep_alive true
+    log_path var/"log/server-go.log"
+    error_log_path var/"log/server-go.log"
```
Comment (SMillerDev) [inline Formula/openiothub-server.rb:30]:
```suggestion
    log_path var/"log/openiothub-server.log"
    error_log_path var/"log/openiothub-server.log"
```

File: Formula/openiothub-server.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/server-go -v 2>&1")
+    assert_match "config created", shell_output("#{bin}/server-go init --config=server.yml 2>&1")
```
Comment (SMillerDev) [inline Formula/openiothub-server.rb:35]:
```suggestion
    assert_match version.to_s, shell_output("#{bin}/openiothub-server -v 2>&1")
    assert_match "config created", shell_output("#{bin}/openiothub-server init --config=server.yml 2>&1")
```

File: Formula/openiothub-server.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.commit=#{Utils.git_head}
+      -X main.date=#{time.iso8601}
+      -X main.builtBy=homebrew
```
Comment (SMillerDev) [inline Formula/openiothub-server.rb:18]:
```suggestion
      -X main.builtBy=#{tap.user}
```

File: Formula/openiothub-server.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    ]
+
+    (etc/"server-go").mkpath
+    system "go", "build", "-mod=vendor", *std_go_args(ldflags: ldflags)
+    etc.install "server-go.yaml" => "server-go/server-go.yaml"
+  end
+
+  service do
+    run [opt_bin/"server-go", "-c", etc/"server-go/server-go.yaml"]
```
[the remark this answers is not in the document]
Comment (dawidd6) [reply Formula/openiothub-server.rb]: pkg_etc is undefined for service block apparently

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**